index.js

```diff
@@ -8,17 +8,38 @@ import isUrl from 'is-url'
 import {visitParents} from 'unist-util-visit-parents'
 import {is} from 'unist-util-is'
 
-const isImgExt = (/** @type {string} */ value) =>
-  /\.(svg|png|jpg|jpeg|gif)$/.test(value)
+/**
+ * @typedef Options
+ * @property {Array<string>} [imageExtensions] file extensions of images
+ */
+
 const isImgPath = (/** @type {string} */ value) =>
   value.startsWith('/') || value.startsWith('./') || value.startsWith('../')
 
+/**
+ * Extensions recognized as images by default
+ */
+export const defaultImageExtensions = [
+  'svg',
+  'png',
+  'jpg',
+  'jpeg',
+  'gif',
+  'webp',
+  'avif'
+]
+
 /**
  * Plugin to add an improved image syntax.
  *
- * @type {import('unified').Plugin<void[], Root>}
+ * @type {import('unified').Plugin<[Options?], Root>}
  */
-export default function remarkImages() {
+export default function remarkImages({
+  imageExtensions = defaultImageExtensions
+} = {}) {
+  const isImgExt = (/** @type {string} */ value) =>
+    new RegExp(`\\.(${imageExtensions.join('|')})$`).test(value)
```

Comment:
I’d recommend first making the regex, as it doesn’t change. Then using the prebuilt regex in the function for each URL

Reply:
added